<div class="alert alert-block alert-info">
<center>PROJETO I</center>
<center>ESTATÍSTICA APLICADA</center>
<center>ALUNO: Fabricio Moreno da Silva</center>
</div>

In [2]:
import pandas as pd

In [3]:
df2018 = pd.read_csv(r"C:\\Users\\yagam\\Downloads\\datatran2018.csv",
                     sep=";",
                     encoding="latin1",
                     low_memory=False)
df2018

id data_inversa     dia_semana   horario  uf     br     km  \
0      100027.0   2018-01-01  segunda-feira  06:00:00  TO  153.0  393,1   
1      100044.0   2018-01-01  segunda-feira  08:30:00  SE  101.0  123,4   
2      100046.0   2018-01-01  segunda-feira  08:50:00  TO  153.0    347   
3      100052.0   2018-01-01  segunda-feira  05:30:00  PA  155.0    221   
4      100053.0   2018-01-01  segunda-feira  10:25:00  BA  116.0  289,6   
...         ...          ...            ...       ...  ..    ...    ...   
69327  412168.0   2018-02-26  segunda-feira  18:40:00  RO  364.0    424   
69328  422003.0   2018-12-15         sábado  09:13:00  PE  101.0     96   
69329  448049.0   2018-12-23        domingo  07:00:00  MG  381.0  825,1   
69330  453741.0   2018-10-21        domingo  17:50:00  RS  285.0    373   
69331  454847.0   2018-09-05   quarta-feira  18:15:00  PB  230.0   16,8   

                     municipio  \
0                 RIO DOS BOIS   
1            ITAPORANGA DAJUDA   
2                       GUARAI   
3          ELDORADO DO CARAJAS   
4                       TUCANO   
...                        ...   
69327                     JARU   
69328  CABO DE SANTO AGOSTINHO   
69329                  CAREACU   
69330         SALDANHA MARINHO   
69331              JOAO PESSOA   

                                          causa_acidente  \
0                            Falta de Atenção à Condução   
1                     Não guardar distância de segurança   
2                            Falta de Atenção à Condução   
3                                     Ingestão de Álcool   
4                                 Ultrapassagem Indevida   
...                                                  ...   
69327  Desobediência às normas de trânsito pelo condutor   
69328  Desobediência às normas de trânsito pelo condutor   
69329                        Defeito Mecânico no Veículo   
69330  Desobediência às normas de trânsito pelo condutor   
69331                 Não guardar distância de segurança   

                   tipo_acidente  ... feridos_graves ilesos ignorados feridos  \
0                Colisão frontal  ...              1      1         0       1   
1               Colisão traseira  ...              0      2         0       0   
2                Colisão lateral  ...              0      2         0       0   
3                Colisão lateral  ...              0      1         0       0   
4      Saída de leito carroçável  ...              1      0         0       1   
...                          ...  ...            ...    ...       ...     ...   
69327            Colisão lateral  ...              0      1         0       2   
69328            Colisão frontal  ...              0      4         0       1   
69329  Saída de leito carroçável  ...              0      0         0       4   
69330            Colisão frontal  ...              1      0         0       1   
69331           Colisão traseira  ...              0      2         0       0   

      veiculos      latitude     longitude  regional  delegacia  \
0            2   -9,35641407  -48,54610294   SPRF-TO   DEL02-TO   
1            2  -11,04476007     -37,36498   SPRF-SE   DEL02-SE   
2            2     -8,952295    -48,500107   SPRF-TO   DEL02-TO   
3            2     -6,345512    -49,389626   SPRF-PA   DEL03-PA   
4            1  -11,08661652  -38,83420181   SPRF-BA   DEL02-BA   
...        ...           ...           ...       ...        ...   
69327        2      -10,4359     -62,47446   SPRF-RO   DEL03-RO   
69328        2   -8,25245569  -35,02595901   SPRF-PE   DEL01-PE   
69329        1      -22,1056      -45,7317   SPRF-MG   DEL16-MG   
69330        2  -28,41949989  -53,12244203   SPRF-RS   DEL10-RS   
69331        2   -7,10997941  -34,84313697   SPRF-PB   DEL01-PB   

                    uop  
0        UOP02-DEL02-TO  
1        UOP01-DEL02-SE  
2        UOP02-DEL02-TO  
3        UOP01-DEL03-PA  
4      DEL10/2-UOP02/BA  
...                 ...  
69327    UOP02-DEL03-R

In [4]:
df2019 = pd.read_csv(r"C:\\Users\\yagam\\Downloads\\datatran2019.csv",
                     sep=";",
                     encoding="latin1",
                     low_memory=False)
df2019

id data_inversa   dia_semana   horario  uf     br     km  \
0      182256.0   2019-01-01  terça-feira  04:00:00  CE  116.0  136,9   
1      182263.0   2019-01-01  terça-feira  05:00:00  MT  158.0  599,5   
2      182277.0   2019-01-01  terça-feira  10:00:00  PA   10.0     34   
3      182289.0   2019-01-01  terça-feira  08:30:00  BA  101.0     13   
4      182307.0   2019-01-01  terça-feira  13:50:00  BA  116.0    440   
...         ...          ...          ...       ...  ..    ...    ...   
67551  444933.0   2019-09-10  terça-feira  17:15:00  TO  153.0    410   
67552  459923.0   2019-06-15       sábado  05:30:00  SP  116.0  209,8   
67553  460497.0   2019-04-20       sábado  01:00:00  RJ  101.0     67   
67554  465497.0   2019-03-15  sexta-feira  10:30:00  RN  427.0     99   
67555  465626.0   2019-04-21      domingo  18:50:00  MG   40.0  529,8   

                   municipio  \
0                     RUSSAS   
1                   AGUA BOA   
2                 DOM ELISEU   
3                   RIO REAL   
4           FEIRA DE SANTANA   
...                      ...   
67551              MIRANORTE   
67552              GUARULHOS   
67553  CAMPOS DOS GOYTACAZES   
67554                  CAICO   
67555               CONTAGEM   

                                          causa_acidente  \
0                                       Animais na Pista   
1                            Defeito Mecânico no Veículo   
2                                Velocidade Incompatível   
3                                     Ingestão de Álcool   
4                                     Ingestão de Álcool   
...                                                  ...   
67551                        Falta de Atenção à Condução   
67552                       Falta de Atenção do Pedestre   
67553                                 Ingestão de Álcool   
67554                        Falta de Atenção à Condução   
67555  Desobediência às normas de trânsito pelo condutor   

                   tipo_acidente  ... feridos_graves ilesos ignorados feridos  \
0        Atropelamento de Animal  ...              0      1         0       0   
1                       Incêndio  ...              0      1         0       0   
2               Colisão traseira  ...              0      6         0       0   
3               Colisão traseira  ...              0      1         0       1   
4      Saída de leito carroçável  ...              1      1         0       2   
...                          ...  ...            ...    ...       ...     ...   
67551  Saída de leito carroçável  ...              0      0         0       1   
67552  Atropelamento de Pedestre  ...              1      1         0       1   
67553  Saída de leito carroçável  ...              0      0         0       2   
67554        Colisão transversal  ...              1      1         1       1   
67555            Colisão lateral  ...              0      1         1       1   

      veiculos      latitude     longitude  regional  delegacia  \
0            1     -4,766018    -38,056034   SPRF-CE   DEL03-CE   
1            1       -14,319       -52,169   SPRF-MT   DEL07-MT   
2            2   -4,15296734  -47,54608154   SPRF-PA   DEL02-PA   
3            2      -11,5896      -37,8761   SPRF-SE   DEL02-SE   
4            2       -12,344       -39,087   SPRF-BA   DEL02-BA   
...        ...           ...           ...       ...        ...   
67551        1   -9,49885975  -48,58649141   SPRF-TO   DEL02-TO   
67552        1  -23,42511799  -46,40221101   SPRF-SP   DEL01-SP   
67553        1  -21,76751094  -41,34876249   SPRF-RJ   DEL08-RJ   
67554        2   -6,45929918  -37,09847983   SPRF-RN   DEL03-RN   
67555        2  -19,91635592  -44,03966566   SPRF-MG   DEL01-MG   

                  uop  
0      UOP01-DEL03-CE  
1      UOP02-DEL07-MT  
2      UOP02-DEL02-PA  
3      UOP03-DEL02-SE  
4      UOP01-DEL02-BA  
...               ...  
67551  UOP02-DEL02-TO  
67552  UOP01-DEL01-SP  
67553  UOP02-DEL08-RJ  
67554  UOP01

In [27]:
df2020 = pd.read_csv(r"C:\\Users\\yagam\\Downloads\\datatran2020.csv",
                     sep=";",
                     encoding="latin1",
                     low_memory=False)
df2020

id data_inversa    dia_semana   horario  uf     br     km  \
0      260068.0   2020-01-01  quarta-feira  05:40:00  PA  316.0     84   
1      260073.0   2020-01-01  quarta-feira  06:00:00  MG  262.0    804   
2      260087.0   2020-01-01  quarta-feira  06:00:00  BA  116.0    191   
3      260116.0   2020-01-01  quarta-feira  10:08:00  SP  116.0     71   
4      260129.0   2020-01-01  quarta-feira  12:10:00  MG  262.0  380,9   
...         ...          ...           ...       ...  ..    ...    ...   
63571  462010.0   2020-12-24  quinta-feira  14:30:00  ES  101.0  270,1   
63572  462426.0   2020-09-06       domingo  23:25:00  MG  262.0    397   
63573  462434.0   2020-10-06   terça-feira  13:10:00  PR  487.0    178   
63574  462484.0   2020-10-04       domingo  17:15:00  GO  153.0    538   
63575  463900.0   2020-02-18   terça-feira  11:45:00  BA  101.0    315   

                       municipio  \
0          SAO FRANCISCO DO PARA   
1                        UBERABA   
2                        CANUDOS   
3                      APARECIDA   
4                        JUATUBA   
...                          ...   
63571                      SERRA   
63572              PARA DE MINAS   
63573               CAMPO MOURAO   
63574                HIDROLANDIA   
63575  PRESIDENTE TANCREDO NEVES   

                                          causa_acidente  \
0                            Falta de Atenção à Condução   
1                            Falta de Atenção à Condução   
2                                      Condutor Dormindo   
3                     Não guardar distância de segurança   
4                                      Condutor Dormindo   
...                                                  ...   
63571  Desobediência às normas de trânsito pelo condutor   
63572  Desobediência às normas de trânsito pelo condutor   
63573  Desobediência às normas de trânsito pelo condutor   
63574  Condutor deixou de manter distância do veículo...   
63575                             Ultrapassagem Indevida   

                   tipo_acidente  ... feridos_graves ilesos ignorados feridos  \
0      Saída de leito carroçável  ...              0      0         0       2   
1            Colisão transversal  ...              0      3         0       1   
2      Saída de leito carroçável  ...              0      0         0       0   
3               Colisão traseira  ...              0      1         0       2   
4      Saída de leito carroçável  ...              0      0         0       1   
...                          ...  ...            ...    ...       ...     ...   
63571        Colisão transversal  ...              1      1         0       1   
63572  Saída de leito carroçável  ...              0      1         0       0   
63573        Colisão transversal  ...              1      0         0       3   
63574           Colisão traseira  ...              0      1         0       1   
63575  Saída de leito carroçável  ...              1      1         0       3   

      veiculos   latitude  longitude  regional  delegacia             uop  
0            1  -1.310193 -47.744564   SPRF-PA   DEL01-PA  UOP02-DEL01-PA  
1            2 -19.767475 -47.987255   SPRF-MG   DEL13-MG  UOP01-DEL13-MG  
2            1 -10.320021 -39.064252   SPRF-BA   DEL07-BA  UOP02-DEL07-BA  
3            2 -22.856517 -45.231143   SPRF-SP   DEL08-SP  UOP01-DEL08-SP  
4            1 -19.947864 -44.381226   SPRF-MG   DEL01-MG  UOP03-DEL01-MG  
...        ...        ...        ...       ...        ...             ...  
63571        2 -20.226621 -40.270751   SPRF-ES   DEL02-ES  UOP01-DEL02-ES  
63572        1 -19.904836 -44.515182   SPRF-MG   DEL01-MG  UOP03-DEL01-MG  
63573        2 -24.041619 -52.404007   SPRF-PR   DEL09-PR  UOP03-DEL09-PR  
63574        2 -17.036500 -49.224300   SPRF-GO   DEL01-GO  UOP03-DEL01-GO  
63575        1 -13.299060 -39.340040   SPRF-BA   DEL05-BA  UOP03-DEL05-BA  

[63576 rows x 30 columns]

In [119]:
len(df2018)

69295

In [120]:
len(df2019)

67446

In [121]:
len(df2020)

63548

<div class="alert alert-block alert-info">
<div>2.</div>
<div>a) Como cada linha corresponde à um acidente, pode se dizer que no ano de 2018, ocorreram 69295 acidentes, em 2019, 67446 acidentes e em 2020, 63548 acidentes, coisa que pode ser confirmada após o uso da função 'len'</div>
<br>
<div>b) Considerando que cada coluna corresponde à uma variável diferente, em cada DataFrame, existem 30 variáveis diferentes</div>
</div>

In [122]:
dftotal = df2018.append(df2019)
dftotal = dftotal.append(df2020)
dftotal
#usado para todas as questões que necessitam analisar o conjunto dos 3 anos juntos

id data_inversa     dia_semana   horario  uf     br     km  \
0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...         ...          ...            ...       ...  ..    ...    ...   
63543  364615.0   2020-03-20    sexta-feira  09:00:00  RN  304.0  171,8   
63544  364810.0   2020-09-12         sábado  23:50:00  BA  116.0   11,5   
63545  364817.0   2020-10-30    sexta-feira  10:30:00  BA  116.0   14,3   
63546  364835.0   2020-09-20        domingo  23:00:00  BA  116.0      8   
63547  364903.0   2020-10-31         sábado  02:10:00  BA  116.0      3   

                  municipio                      causa_acidente  \
0                   RESENDE                   Condutor Dormindo   
1             FLORIANOPOLIS  Não guardar distância de segurança   
2                  ITABORAI              Ultrapassagem Indevida   
3                   SARANDI                  Ingestão de Álcool   
4                  CANDIOTA         Falta de Atenção à Condução   
...                     ...                                 ...   
63543               ANGICOS         Falta de Atenção à Condução   
63544  VITORIA DA CONQUISTA                  Ingestão de Álcool   
63545  VITORIA DA CONQUISTA         Falta de Atenção à Condução   
63546  VITORIA DA CONQUISTA             Velocidade Incompatível   
63547  VITORIA DA CONQUISTA                    Animais na Pista   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0         Saída de leito carroçável  ...              4      1         0   
1                  Colisão traseira  ...              2      1         0   
2                   Colisão frontal  ...              0      3         1   
3               Colisão transversal  ...              0      2         0   
4         Saída de leito carroçável  ...              1      0         0   
...                             ...  ...            ...    ...       ...   
63543              Colisão traseira  ...              1      2         0   
63544               Colisão lateral  ...              0      2         0   
63545               Colisão lateral  ...              0      2         0   
63546  Queda de ocupante de veículo  ...              0      0         0   
63547       Atropelamento de Animal  ...              1      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...       ...      ...           ...           ...       ...        ...   
63543       2        4   -5,68711053    -36,419458     SR-RN      UOP03   
63544       1        2  -14,87308214  -40,87141514     SR-BA      UOP01   
63545       2        2  -14,89597653  -40,86098671     SR-BA      UOP01   
63546       1        1  -14,84732305  -40,88634968     SR-BA      UOP01   
63547       1        1    -14,824848    -40,861311     SR-BA      UOP01   

            uop  
0      UOP03/RJ  
1      UOP01/SC  
2      UOP02/RJ  
3      UOP01/RS  
4      UOP03/RS  
...         ...  
63543     DEL01  
63544     DEL08  
63545     DEL08  
63546     DEL08  
63547     DEL08  

[200289 rows x 30 columns]

In [123]:
dftotal['municipio'].value_counts()

CURITIBA                3063
BRASILIA                2997
SAO JOSE                2322
GUARULHOS               2111
PALHOCA                 1887
                        ... 
ARAGUAIANA                 1
SANTA MARIA MADALENA       1
NORTELANDIA                1
SANTA CRUZ CABRALIA        1
URANDI                     1
Name: municipio, Length: 1926, dtype: int64

<div class="alert alert-block alert-info">
<div>2.</div>
c) Como se pode perceber pelos dados acima, trazidos pelo 'dftotal', as 5 cidades brasileiras aonde mais ocorreram acidentes em rodovias federais são: Curitiba, Brasília, São José, Guarulhos e Palhoça
</div>

In [124]:
dfPB2019 = ((df2019['feridos_graves']!=0)&(df2019['uf']=='PB')).sum()
dfPB2019

414

<div class="alert alert-block alert-info">
<div>2.</div>
d) Como é perceptível pelo resultado trazido pelo 'dfPB2019', no ano de 2019, na Paraíba, ocorreram 414 acidentes que deixaram pelo menos 1 ferido grave
</div>

In [125]:
rankingUF = pd.DataFrame(dftotal['uf'].value_counts())
rankingUF

uf
MG  26160
SC  24145
PR  22850
RJ  13417
RS  13216
SP  12936
BA  10482
GO  10046
ES   8018
PE   8011
MT   6963
CE   4962
RO   4648
PB   4553
MS   4505
RN   4190
PI   3920
MA   3501
PA   3040
DF   2997
AL   1948
SE   1746
TO   1704
AC    789
RR    723
AP    456
AM    363

In [126]:
rankingUF.to_csv(index=True)

',uf\nMG,26160\nSC,24145\nPR,22850\nRJ,13417\nRS,13216\nSP,12936\nBA,10482\nGO,10046\nES,8018\nPE,8011\nMT,6963\nCE,4962\nRO,4648\nPB,4553\nMS,4505\nRN,4190\nPI,3920\nMA,3501\nPA,3040\nDF,2997\nAL,1948\nSE,1746\nTO,1704\nAC,789\nRR,723\nAP,456\nAM,363\n'

<div class="alert alert-block alert-info">
<div>3.</div>
a) À partir do DataFrame 'rankingUF' criado, o qual também foi codificado em .csv logo em seguida, podemos ver o ranking dos estados brasileiros em que mais ocorreram acidentes do ano de 2018 à 2020
</div>

In [127]:
rankingDIAS = pd.DataFrame(dftotal['dia_semana'].value_counts())
rankingDIAS

dia_semana
sábado              33005
domingo             33001
sexta-feira         30777
segunda-feira       27075
quinta-feira        26229
quarta-feira        25353
terça-feira         24849

In [128]:
rankingDIAS.to_csv(index=True)

',dia_semana\nsábado,33005\ndomingo,33001\nsexta-feira,30777\nsegunda-feira,27075\nquinta-feira,26229\nquarta-feira,25353\nterça-feira,24849\n'

<div class="alert alert-block alert-info">
<div>3.</div>
b) Como também foi feito no DatFrame do item anterior, o agora criado, denominado de 'rankingDIAS', também foi formatado para .csv e nos mostra um ranking, mas desta vez, dos dias da semana, no período de 2018 à 2020, em que mais ocorreram acidentes
</div>

In [129]:
causasAcidentes = pd.DataFrame(dftotal['causa_acidente'].value_counts())
causasAcidentes

causa_acidente
Falta de Atenção à Condução                                  73231
Desobediência às normas de trânsito pelo condutor            24089
Velocidade Incompatível                                      18763
Ingestão de Álcool                                           15689
Não guardar distância de segurança                           12082
Defeito Mecânico no Veículo                                  11190
Condutor Dormindo                                             7225
Pista Escorregadia                                            6752
Falta de Atenção do Pedestre                                  5685
Animais na Pista                                              4551
Ultrapassagem Indevida                                        3590
Defeito na Via                                                3059
Avarias e/ou desgaste excessivo no pneu                       2712
Mal Súbito                                                    1976
Restrição de Visibilidade                                     1515
Objeto estático sobre o leito carroçável                      1422
Agressão Externa                                              1082
Desobediência às normas de trânsito pelo pedestre             1053
Carga excessiva e/ou mal acondicionada                        1049
Ingestão de álcool e/ou substâncias psicoativas...            1048
Sinalização da via insuficiente ou inadequada                  780
Fenômenos da Natureza                                          761
Deficiência ou não Acionamento do Sistema de Il...             533
Ingestão de Substâncias Psicoativas                            253
Reação tardia ou ineficiente do condutor                        24
Ausência de reação do condutor                                  21
Condutor deixou de manter distância do veículo ...              21
Manobra de mudança de faixa                                     19
Transitar na contramão                                          14
Demais falhas mecânicas ou elétricas                            13
Acessar a via sem observar a presença dos outro...              13
Ingestão de álcool pelo condutor                                 8
Desrespeitar a preferência no cruzamento                         8
Conversão proibida                                               6
Chuva                                                            6
Trafegar com motocicleta (ou similar) entre as ...               5
Pedestre andava na pista                                         4
Condutor desrespeitou a iluminação vermelha do ...               4
Falta de acostamento                                             3
Acesso irregular                                                 3
Transitar no acostamento                                         3
Acumulo de água sobre o pavimento                                2
Demais falhas na via                                             2
Mal súbito do condutor                                           2
Afundamento ou ondulação no pavimento                            2
Iluminação deficiente                                            2
Entrada inopinada do pedestre                                    2
Acumulo de areia ou detritos sobre o pavimento                   1
Pista esburacada                                                 1
Estacionar ou parar em local proibido                            1
Fumaça                                                           1
Pedestre cruzava a pista fora da faixa                           1
Acostamento em desnível                                          1
Condutor usando celular                                          1
Frear bruscamente                                                1
Ausência de sinalização                                          1
Acumulo de óleo sobre o pavimento                                1
Ingestão de álcool ou de substâncias psicoativa...               1
Curva acentuada                                                  1

In [130]:
causasAcidentes.head(1)

causa_acidente
Falta de Atenção à Condução           73231

In [131]:
causasRaras = causasAcidentes.query('causa_acidente == 1')
causasRaras

causa_acidente
Acumulo de areia ou detritos sobre o pavimento                   1
Pista esburacada                                                 1
Estacionar ou parar em local proibido                            1
Fumaça                                                           1
Pedestre cruzava a pista fora da faixa                           1
Acostamento em desnível                                          1
Condutor usando celular                                          1
Frear bruscamente                                                1
Ausência de sinalização                                          1
Acumulo de óleo sobre o pavimento                                1
Ingestão de álcool ou de substâncias psicoativa...               1
Curva acentuada                                                  1

<div class="alert alert-block alert-info">
<div>4.</div>
    a) Como é visto usando a função "head", e usando o parâmetro 1 para determinar a primeira linha do DataFrame, podemos ver que a causa mais frequente para a ocorrência de acidentes é a 'falta de atenção à condução', enquanto as causas mais raras, empatadas com apenas uma ocorrência de cada durante os 3 anos (2018-20), podem ser vistas com o auxílio da função "query"
<div>

In [132]:
df2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69295 entries, 0 to 69294
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      69295 non-null  float64
 1   data_inversa            69295 non-null  object 
 2   dia_semana              69295 non-null  object 
 3   horario                 69295 non-null  object 
 4   uf                      69295 non-null  object 
 5   br                      69181 non-null  float64
 6   km                      69181 non-null  object 
 7   municipio               69295 non-null  object 
 8   causa_acidente          69295 non-null  object 
 9   tipo_acidente           69295 non-null  object 
 10  classificacao_acidente  69295 non-null  object 
 11  fase_dia                69295 non-null  object 
 12  sentido_via             69295 non-null  object 
 13  condicao_metereologica  69295 non-null  object 
 14  tipo_pista              69295 non-null

In [133]:
df2018['data_inversa'] = pd.to_datetime(df2018['data_inversa'])
df2019['data_inversa'] = pd.to_datetime(df2019['data_inversa'])
df2020['data_inversa'] = pd.to_datetime(df2020['data_inversa'])
df2018.info()
df2019.info()
df2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69295 entries, 0 to 69294
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   id                      69295 non-null  float64       
 1   data_inversa            69295 non-null  datetime64[ns]
 2   dia_semana              69295 non-null  object        
 3   horario                 69295 non-null  object        
 4   uf                      69295 non-null  object        
 5   br                      69181 non-null  float64       
 6   km                      69181 non-null  object        
 7   municipio               69295 non-null  object        
 8   causa_acidente          69295 non-null  object        
 9   tipo_acidente           69295 non-null  object        
 10  classificacao_acidente  69295 non-null  object        
 11  fase_dia                69295 non-null  object        
 12  sentido_via             69295 non-null  object

<div class="alert alert-block alert-info">
Os passos feitos logo antes desse comentário foram necessários para que a coluna "data_inversa" mudassem de 'object' para 'datetime', e o que vem a ser realizado logo à seguir, possa ser feito com êxito
</div>

In [134]:
agrupamento2018 = df2018.groupby(df2018['data_inversa'].dt.month)['feridos_graves'].sum()
agrupamento2018final = pd.DataFrame(agrupamento2018)
agrupamento2018final['ilesos'] = df2018.groupby(df2018['data_inversa'].dt.month)['ilesos'].sum()
totalFeridos = df2018.groupby(df2018['data_inversa'].dt.month)['feridos'].sum()
agrupamento2018final['total_pessoas'] = df2018.groupby(df2018['data_inversa'].dt.month)['pessoas'].sum()
agrupamento2018final['proporção_ilesos/total'] = agrupamento2018final['ilesos']/agrupamento2018final['total_pessoas']
agrupamento2018final['proporção_graves/total'] = agrupamento2018final['feridos_graves']/agrupamento2018final['total_pessoas']
agrupamento2018final

feridos_graves  ilesos  total_pessoas  proporção_ilesos/total  \
data_inversa                                                                  
1                       1510    8554          16665                0.513291   
2                       1295    6999          14047                0.498256   
3                       1539    7276          15110                0.481535   
4                       1492    5691          13230                0.430159   
5                       1315    4851          11489                0.422230   
6                       1498    5471          12870                0.425097   
7                       1502    5958          13618                0.437509   
8                       1359    5456          12776                0.427051   
9                       1504    5480          13027                0.420665   
10                      1452    5545          12890                0.430178   
11                      1523    5719          13197                0.433356   
12                      1717    6876          15883                0.432916   

              proporção_graves/total  
data_inversa                          
1                           0.090609  
2                           0.092191  
3                           0.101853  
4                           0.112774  
5                           0.114457  
6                           0.116395  
7                           0.110295  
8                           0.106371  
9                           0.115453  
10                          0.112645  
11                          0.115405  
12                          0.108103

In [135]:
agrupamento2019 = df2019.groupby(df2019['data_inversa'].dt.month)['feridos_graves'].sum()
agrupamento2019final = pd.DataFrame(agrupamento2019)
agrupamento2019final['ilesos'] = df2019.groupby(df2019['data_inversa'].dt.month)['ilesos'].sum()
totalFeridos = df2019.groupby(df2019['data_inversa'].dt.month)['feridos'].sum()
agrupamento2019final['total_pessoas'] = df2019.groupby(df2019['data_inversa'].dt.month)['pessoas'].sum()
agrupamento2019final['proporção_ilesos/total'] = agrupamento2019final['ilesos']/agrupamento2019final['total_pessoas']
agrupamento2019final['proporção_graves/total'] = agrupamento2019final['feridos_graves']/agrupamento2019final['total_pessoas']
agrupamento2019final

feridos_graves  ilesos  total_pessoas  proporção_ilesos/total  \
data_inversa                                                                  
1                       1485    5562          13111                0.424224   
2                       1260    5004          11799                0.424104   
3                       1555    5878          13569                0.433193   
4                       1358    5327          12815                0.415685   
5                       1545    5312          12795                0.415162   
6                       1631    5683          13411                0.423757   
7                       1672    6260          14331                0.436815   
8                       1566    5544          13421                0.413084   
9                       1514    5761          13682                0.421064   
10                      1669    5836          13931                0.418922   
11                      1535    5886          13823                0.425812   
12                      1783    6581          15585                0.422265   

              proporção_graves/total  
data_inversa                          
1                           0.113264  
2                           0.106789  
3                           0.114599  
4                           0.105970  
5                           0.120750  
6                           0.121617  
7                           0.116670  
8                           0.116683  
9                           0.110656  
10                          0.119805  
11                          0.111047  
12                          0.114405

<div class="alert alert-block alert-info">
<div>4.</div>
    b) Como podem ser vistos nos três dataframes anteriores, eles apresentam, de 2018 à 2020, a quantidade de feridos graves, pessoas ilesas e o total de pessoas envolvidas em acidentes em cada mês durante os anos, bem como apresenta como pedido, as proporções de pessoas ilesas para o total de pessoas, como também a proporção de feridos graves para a quantidade do total de pessoas
</div>

In [136]:
df2020_2 = df2020
df2020_2['meses'] = df2020_2['data_inversa'].dt.month
pos_pandemia = (df2020_2['meses']>= 3).sum()
pos_pandemia

52584

In [137]:
df2020_3 = df2020
df2020_3['meses'] = df2020_3['data_inversa'].dt.month
pre_pandemia = (df2020_3['meses']< 3).sum()
pre_pandemia

10964

In [138]:
#Soma total de acidentes pré-pandemia
total_pre = len(df2018) + len(df2019)
total_pre = total_pre + pre_pandemia
total_pre 

147705

<div class="alert alert-block alert-info">
Nos passos anteriores, foi separado, primeiro, a quantidade de acidentes ocorridos à partir de Março de 2020, momento em que o estado pandêmico foi estabelecido no Brasil<br>
Após isso, foram coletadas a quantidade de acidentes ocorridos em Janeiro e Fevereiro de 2020, meses em que ainda não havia pandemia no Brasil, e com isso, essa quantidade encontrada foi somada à quantidade total de acidentes nos anos anteriores
</div>

In [139]:
#Proporções pedidas
QTD_MESES_PRE = 26 #Quantidade de meses no período pré-pandemia
QTD_MESES_POS = 10 #Quantidade de meses no período pós-pandemia
media_pos = "{:.2f}".format(pos_pandemia/10)
media_pre = "{:.2f}".format(total_pre/26)
print("média de acidentes, por mês, com a pandemia 'ativa':", media_pos)
print("média de acidentes, por mês, sem a pandemia 'ativa':", media_pre)

média de acidentes, por mês, com a pandemia 'ativa': 5258.40
média de acidentes, por mês, sem a pandemia 'ativa': 5680.96


<div class="alert alert-block alert-info">
Nas divisões anteriores, temos a proporção, ou média, de acidentes que ocorreram, por mês, nos períodos pré e pós pandêmicos, os quais serão apresentados também no DataFrame à seguir.
</div>

In [140]:
total_acidentes = [total_pre, pos_pandemia]
medias = [media_pre, media_pos]

dfMedias = pd.DataFrame({'total_acidentes':total_acidentes, 
                         'proporção':medias})
dfMedias

total_acidentes proporção
0           147705   5680.96
1            52584   5258.40

<div class="alert alert-block alert-info">
<div>4.</div>
    c) Com base no DataFrame anterior, onde a linha 0 se refere ao período pré-pandêmico, e a linha 1 ao período pós-pandêmico, é possível analisar e perceber que a pandemia reduziu sim a frequência de acidentes ocorridos nas rodovias federais
</div>